### PR TITLE
New Lambda docs

### PR DESCRIPTION
**Pull request by robfusion**

### CODE REVIEW

File: site/docs/v1/tech/lambdas/scim-user-request-converter.adoc [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+** `options.disableDomainBlock`
+** `options.sendSetPasswordEmail`
+** `options.skipVerification` NOTE: sendSetPasswordEmail is only utilized during a User create request
+* `scimUser` - the SCIM request object
+
+The FusionAuth objects are well documented here in the link:/docs/v1/tech/apis/users[User API] documentation. The SCIM User object is a JavaScript object containing the SCIM User and optionally the SCIM EnterpriseUser request JSON payload. See link:https://datatracker.ietf.org/doc/html/rfc7643#section-4.1[SCIM User] and link:https://datatracker.ietf.org/doc/html/rfc7643#section-4.3[SCIM EnterpriseUser extension].
+
+You may add or modify anything in the `user` and `options` objects.
+
+== Assigning the lambda
```
Comment (mooreds) [inline site/docs/v1/tech/lambdas/scim-user-request-converter.adoc:36]:
```suggestion
== Assigning the Lambda
```

File: site/docs/v1/tech/lambdas/scim-user-request-converter.adoc [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+Once a lambda is created, you must assign it to a Tenant. See the SCIM tab in the Tenant configuration.
+
+== Default lambda
```
Comment (mooreds) [inline site/docs/v1/tech/lambdas/scim-user-request-converter.adoc:40]:
```suggestion
== Default Lambda
```

File: site/docs/v1/tech/lambdas/scim-user-response-converter.adoc [hunk 3]
```diff
@@ -0,0 +1,78 @@
+---
+layout: doc
+title: SCIM User Response Converter lambda
+description: An overview of the SCIM User Response Converter lambda
+---
+
+:sectnumlevels: 0
+
+== SCIM User Response Converter lambda
```
Comment (mooreds) [inline site/docs/v1/tech/lambdas/scim-user-response-converter.adoc:9]:
```suggestion
== SCIM User Response Converter Lambda
```

File: site/docs/v1/tech/lambdas/scim-user-response-converter.adoc [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+If you would like to convert an outgoing FusionAuth User response into a SCIM User and optionally SCIM EnterpriseUser, you can specify a lambda in the SCIM configuration. This lambda will be invoked after the User was acted upon.
```
Comment (mooreds) [inline site/docs/v1/tech/lambdas/scim-user-response-converter.adoc:11]: You have to specify this lambda.

File: site/docs/v1/tech/lambdas/scim-user-response-converter.adoc [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+This lambda must contain a function named `convert` that takes two parameters. The parameters that the lambda is passed are:
+
+* `scimUser` - the SCIM response object
+* `user` - the FusionAuth User object
+
+The FusionAuth objects is well documented here in the link:/docs/v1/tech/apis/users[User API] documentation. The SCIM User object is a JavaScript object containing the SCIM User and optionally the SCIM EnterpriseUser response JSON payload. See link:https://datatracker.ietf.org/doc/html/rfc7643#section-4.1[SCIM User] and link:https://datatracker.ietf.org/doc/html/rfc7643#section-4.3[SCIM EnterpriseUser extension].
+
+You may add or modify anything in the `scimUser` object.
+
+== Assigning the lambda
```
Comment (mooreds) [inline site/docs/v1/tech/lambdas/scim-user-response-converter.adoc:31]:
```suggestion
== Assigning the Lambda
```

File: site/docs/v1/tech/lambdas/scim-user-response-converter.adoc [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+
+Once a lambda is created, you must assign it to a Tenant. See the SCIM tab in the Tenant configuration.
+
+== Default lambda
```
Comment (mooreds) [inline site/docs/v1/tech/lambdas/scim-user-response-converter.adoc:35]:
```suggestion
== Default Lambda
```

**Review by mooreds — CHANGES_REQUESTED** [review]
A few small changes. The only big question I had is, when a request comes into a lambda, does the lambda know what kind of request it is (create, update, delete)? I don't see a way to know that but it seems like it'd be a useful bit of info to know.

File: site/_data/lambdas.yaml [hunk 7]
```diff
@@ -95,3 +95,23 @@
   docLink: /docs/v1/tech/lambdas/xbox-reconcile
   version: 1.28.0
   menuIcon: fa-xbox fab
+- displayName: SCIM Group Request Converter
+  typeText: SCIMGroupRequestConverter
+  docLink: /docs/v1/tech/lambdas/scim-group-request-converter
+  version: 1.36.0
+  menuIcon: fa-server
```
[the remark this answers is not in the document]
Comment (robfusion) [reply site/_data/lambdas.yaml:102]: Hmm, that's weird. I don't know why they would work for me and not you.

**Comment by robfusion** [comment]
> A few small changes. The only big question I had is, when a request comes into a lambda, does the lambda know what kind of request it is (create, update, delete)? I don't see a way to know that but it seems like it'd be a useful bit of info to know.

No, they won't know what kind of request really, but ultimately it shouldn't matter. The purpose of the lambda is just as a converter from SCIM object to FA object or vice versa. I guess we could consider adding another param or something to indicate the HTTP verb.

File: site/_data/lambdas.yaml (hunk 7 above)
[the remark this answers is not in the document]
Comment (robfusion) [reply site/_data/lambdas.yaml:102]: I did have to do an sb clean and then restart to get them to update. Not sure why.

File: site/docs/v1/tech/core-concepts/tenants.adoc [hunk 8]
```diff
@@ -447,6 +447,39 @@ Full documentation on link:/docs/v1/tech/core-concepts/scim/[SCIM] can be found
 
 The SCIM server configuration to enable incoming SCIM client provisioning requests.
 
+===== SCIM settings
+
```
[the remark this answers is not in the document]
Comment (robfusion) [reply site/docs/v1/tech/core-concepts/tenants.adoc:451]: Should be on on line 444

File: site/docs/v1/tech/lambdas/scim-group-response-converter.adoc [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+layout: doc
+title: SCIM Group Response Converter lambda
+description: An overview of the SCIM Group Response Converter lambda
+---
+
+:sectnumlevels: 0
+
+== SCIM Group Response Converter lambda
+
+If you would like to convert an outgoing FusionAuth Group response into a SCIM Group, you can specify a lambda in the SCIM configuration. This lambda will be invoked after the Group was acted upon.
```
[the remark this answers is not in the document]
Comment (robfusion) [reply site/docs/v1/tech/lambdas/scim-group-response-converter.adoc]: fixed

File: site/docs/v1/tech/lambdas/scim-user-request-converter.adoc [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+----
+function convert(user, options, scimUser) {
+  // Lambda code goes here
+}
+----
+
+This lambda must contain a function named `convert` that takes four parameters. The parameters that the lambda is passed are:
+
+* `user` - the FusionAuth User object
+* `options` - options contains optional FusionAuth request options
```
[the remark this answers is not in the document]
Comment (robfusion) [reply site/docs/v1/tech/lambdas/scim-user-request-converter.adoc:25]: done

File: site/docs/v1/tech/lambdas/scim-group-request-converter.adoc [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+----
+function convert(group, members, options, scimGroup) {
+
+  // Un-comment this line to see the scimGroup object printed to the event log
+  // console.info(JSON.stringify(scimGroup, null, 2));
+
+  // Request options
+  // FusionAuth allows you to assign one or more application roles to a group.
+  // To use this feature, assign one or more application Ids here.
+  // options.roleIds = [];
```
[the remark this answers is not in the document]
Comment (robfusion) [reply site/docs/v1/tech/lambdas/scim-group-request-converter.adoc:52]: done

File: site/_data/lambdas.yaml (hunk 7 above)
[the remark this answers is not in the document]
Comment (mooreds) [reply site/_data/lambdas.yaml:102]: Interesting, they still don't appear for me after that.

File: site/docs/v1/tech/lambdas/scim-group-request-converter.adoc [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+----
+
+This lambda must contain a function named `convert` that takes four parameters. The parameters that the lambda is passed are:
+
+* `group` - the FusionAuth Group object
+* `members` - the members in this FusionAuth Group
+* `options` - options.roleIds contains FusionAuth application Ids to assign roles to the Group
+* `scimGroup` - the SCIM request object
+
+The FusionAuth object is well documented here in the link:/docs/v1/tech/apis/groups[Group API] documentation. The SCIM Group object is a JavaScript object containing the SCIM Group request JSON payload. See link:https://datatracker.ietf.org/doc/html/rfc7643#section-4.2[SCIM Group].
```
[the remark this answers is not in the document]
Comment (mooreds) [reply site/docs/v1/tech/lambdas/scim-group-request-converter.adoc:30]: I think just saying "members is basically just a list of FA User Ids" is fine.

File: site/docs/v1/tech/lambdas/scim-user-response-converter.adoc (hunk 4 above)
Comment (mooreds) [reply site/docs/v1/tech/lambdas/scim-user-response-converter.adoc]: ping

File: site/docs/v1/tech/lambdas/scim-group-request-converter.adoc (hunk 12 above)
[the remark this answers is not in the document]
Comment (robfusion) [reply site/docs/v1/tech/lambdas/scim-group-request-converter.adoc:32]: Took another shot at it

**Review by mooreds — APPROVED** [review]

**Pull request merged**